jax/numpy/lax_numpy.py

```diff
@@ -897,6 +897,45 @@ def histogram(a, bins=10, range=None, weights=None, density=None):
     counts = counts / bin_widths / counts.sum()
   return counts, bin_edges
 
+@_wraps(np.histogramdd)
+def histogramdd(sample, bins=10, range=None, weights=None, density=None):
+  _check_arraylike("histogramdd", sample)
+  N, D = shape(sample)
+
+  if weights is not None and weights.shape != (N,):
+    raise ValueError("should have one weight for each sample.")
+
+  bin_idx_by_dim = D*[None]
+  nbins = D*[None]
+  bin_edges_by_dim = D*[None]
+  dedges = D*[None]
+
+  for i in builtins.range(D):
+    bin_edges = histogram_bin_edges(sample[:, i], bins[i], range)
+    bin_idx = searchsorted(bin_edges, sample[:, i], side='right')
+    bin_idx = where(sample[:, i] == bin_edges[-1], bin_idx - 1, bin_idx)
+    bin_idx_by_dim[i] = bin_idx
+    nbins[i] = len(bin_edges) + 1
+    bin_edges_by_dim[i] = bin_edges
+    dedges[i] = diff(bin_edges_by_dim[i])
+
+  xy = ravel_multi_index(bin_idx_by_dim, nbins)
+  hist = bincount(xy, weights, minlength=prod(asarray(nbins)))
```

Comment:
If you're using `jnp.bincount`, you need to pass the `length` parameter in order for it to be compatible with `jit`. See the note here: https://github.com/google/jax/blob/aaa5724eabe8c88036ac3ee808a4bc0714c578ea/jax/numpy/lax_numpy.py#L1496-L1503

Reply:
Thanks yep it looks like that was the problem.  There were also compilation issues associated with the `prod()` call so I changed that numpy.  I also fixed the compilation issues related to your previous comment.  It looks like the tests are passing now.  Thanks again.